### PR TITLE
Add g1_lincomb function

### LINKED ISSUE #16: Cryptol Spec: `g1_lincomb`
Goal: implement the Cryptol function for [`g1_lincomb`](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#g1_lincomb)

Task(s):
- [x] Implement `g1_lincomb` in `BlsHelpers.cry`

Dependencies
- [x] Implement [`bls.bytes48_to_G1` (pointer to `G1_to_pubkey`)](https://github.com/ethereum/py_ecc/blob/main/py_ecc/bls/g2_primitives.py#L54)
   - NOTES: `bls.bytes48_to_G1` points to the `G1_to_pubkey ` function referenced above that wraps [`decompress_G1`](https://github.com/ethereum/py_ecc/blob/main/py_ecc/bls/point_compression.py#L80)
 - [x] Implement [`bls.G1_to_bytes48` (pointer to `pubkey_to_G1`)](https://github.com/ethereum/py_ecc/blob/70c194edd8d1eb1457805442f0162499dbc0aac5/py_ecc/bls/g2_primitives.py#L49)

### PR DESCRIPTION
**Pull request by b13decker**
Add the g1_lincomb function with tests. Fixed a couple of other issues that were exposed when I did this.

Closes #16

### CODE REVIEW

**Review by marsella — APPROVED** [review]
No real concerns; if you decide to do the numeric constraint guard I'd like to take another pass.

Thanks for adding the doc links, I had left a question about them in your other PR so it's nice to see them here now.

File: spec/Spec/BlsHelpers.cry [hunk 1]
```diff
@@ -84,6 +86,22 @@ bytes48_to_G1 bytes = decompress_G1 bytes
...
+/**
+ * A KZG commitment is just a G1 point
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#custom-types
+ */
+type KZGCommitment = Bytes48
+
+/**
+ * BLS multiscalar multiplication in G1. This can be naively implemented using double-and-add.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#g1_lincomb
+ * NOTE: we don't handle the case were there is an empty list of points or scalars provided.
```
[the remark this answers is not in the document]
Comment (b13decker) [reply spec/Spec/BlsHelpers.cry:98]: Excellent comments. I'll do both and then ping you.

[the remark this answers is not in the document]
Comment (b13decker) [reply spec/Spec/BlsHelpers.cry]: @marsella Just added two commits for your two comments. Let me know if those look good. Thanks!

**Review by marsella — APPROVED** [review]
Looks good, thanks for fixing!

**Pull request merged**